Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -372,14 +390,20 @@ void Acts::SurfaceMaterialMapper::mapMaterialTrack(
     // Now assign the material for the accumulation process
     auto tBin = currentAccMaterial->second.accumulate(
         currentPos, rmIter->materialSlab, currentPathCorrection);
-    auto itBin =
-        touchedMapBins.insert(MapBin(&(currentAccMaterial->second), tBin));
+    if (touchedMapBins.count(&(currentAccMaterial->second)) == 0) {
```

Comment:
So previously it was recording them every time even if already in `touchedMapBins`?

Reply:
Yes and no we would add the same bin multiple time to touchedMapBins. This wasn't really an issue because it just meant we would run the averaging on that bin multiple time with one hit (effectively doing nothing except waist time). But it was causing some issue in the newly added variance computation and was a waste of time so I fixed it.